steps/test/unit/tReorderMSData.cc

```diff
@@ -76,19 +108,36 @@ void test1(std::string msPath, std::string msOutPath) {
 BOOST_AUTO_TEST_CASE(testReorder1) {
   test1("../resources/midbands/reorderTestMS.ms", "./testOut.ms");
 
+  // Default stream buffer iterator has a end-of-stream state, so using this to
+  // detect end of the stream.
+  std::istreambuf_iterator<char> endOfStream;
+
   // Data File check
-  BOOST_CHECK(
-      compareBinaries("../resources/midbands/refTmps/reorderTestVis.tmp",
-                      "./testOut.ms-part0000-I-b0.tmp"));
+  std::pair<std::istreambuf_iterator<char>, std::istreambuf_iterator<char>>
+      filePtrPair_vis;
+  filePtrPair_vis =
+      compareBinaryFiles("../resources/midbands/refTmps/reorderTestVis.tmp",
+                         "./testOut.ms-part0000-I-b0.tmp", false);
+  BOOST_REQUIRE_EQUAL_COLLECTIONS(filePtrPair_vis.first, endOfStream,
```

Comment:
Instead of having the function return file pointers and calling `BOOST_REQUIRE_EQUAL_COLLECTIONS`  here, why not just call `BOOST_REQUIRE_EQUAL_COLLECTIONS` inside the function? I think that would probably be cleaner